src/util/updateUrlHistory.js

```diff
@@ -23,7 +25,7 @@ export const updateUrlHistory = (thoughtsRanked = RANKED_ROOT, { replace, though
     return
   }
 
-  const decoded = decodeThoughtsUrl(window.location.pathname, thoughtIndex, contextIndex)
+  const decoded = decodeThoughtsUrl({ thoughtIndex, contextIndex }, window.location.pathname)
```

Comment:
Maybe we should pass the state here too, just like we're doing in every other call to this selector. 
Because this might break or need to be updated in case you update your selector to use more items from the state in the future.

Reply:
Yes, although I have instructed Umair to convert one function per commit, so this should show up in a separate commit.